docs/contribute/api-docs-contribution/guidelines.md

```diff
@@ -0,0 +1,575 @@
+class AlertRule {
+  /**
+   * An ISO-8601 timestamp that indicates when the event was detected.
+   */
+  detectedAt: string
+}
+```
+:::
+
+::::
+
+## Document path parameters
+
+Path parameters are variables in URL paths that users must provide to access specific resources. Proper documentation helps users understand what values are expected and how to construct valid requests.
+
+Here are some principles for documenting path parameters:
+
+- **Write clear summaries and descriptions:** Follow the general guidance for [writing summaries](#write-summaries) and [descriptions](#write-descriptions)
+- **All path parameters are required:** the `required` field must be present and it must be `true` in the OpenAPI doc
+- **Provide alternative paths for optional variations:** If you need optional parameters, create separate endpoint definitions
+- **Document parameter constraints:** Specify valid formats, patterns, or value ranges
+- **Explain parameter relationships:** Clarify how multiple path parameters work together and any dependencies between them
+
+### Examples
+
+::::{tab-set}
+:group: implementations
+
+:::{tab-item} OpenAPI
+:sync: openapi
+
+Define path parameters directly in the path specification:
+
+```yaml
+paths:
+  /users/{userId}/posts/{postId}:
+    get:
+      summary: Get a specific post
+      parameters:
+        - name: userId
+          in: path
+          required: true
+          description: The unique identifier of the user
+          schema:
+            type: string
+            example: "12345"
+        - name: postId
+          in: path
+          required: true
+          description: The unique identifier of the post
+          schema:
+            type: string
+            example: "post-abc"
+```
+
+For optional variations, create separate paths:
+
+```yaml
+paths:
+  /books:
+    get:
+      summary: List all books
+  /books/{category}:
+    get:
+      summary: List books by category
+      parameters:
+        - name: category
+          in: path
+          required: true
+          description: The book category to filter by
+          schema:
+            type: string
+            example: "fiction"
+```
+:::
+
+:::{tab-item} Elasticsearch
+:sync: elasticsearch
+
+For endpoints that can accept optional parameters, define multiple paths in the `urls` array:
+
+```ts
+urls: [
+  {
+    path: "/{index}/_doc/{id}",
+    methods: ["PUT", "POST"]
+  },
+  {
+    path: "/{index}/_doc",
+    methods: ["POST"]
+  }
+]
+```
+
+**Notes:**
+- The `make transform-to-openapi` output shows these two paths as separate entities
+- The `make transform-to-openapi-for-docs` output contains only the most complex variation of the paths (and mentions the others in its description)
+:::
+
+::::
+
+## Document enum values
+
+Enums define a fixed set of allowed values for a property. Well-documented enum values help users understand the purpose of each option and when to use it.
+
+Here are some principles for documenting enum values:
```

Comment:
Maybe good to use the full term at least once?
```suggestion
Here are some principles for documenting enumerated values:
```

Reply:
Cool might do this in opening sentence:  "Enum**erated types** (enums) define a fixed set of allowed values for a property. "